Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialAsset.cpp

```diff
@@ -102,9 +104,19 @@ namespace AZ
 
         AZStd::array_view<MaterialPropertyValue> MaterialAsset::GetPropertyValues() const
         {
-            if (!m_propertyNames.empty() && m_isDirty)
+            if (!m_propertyNames.empty())
             {
-                const_cast<MaterialAsset*>(this)->RealignPropertyValuesAndNames();
+                const uint32_t materialTypeVersion = m_materialTypeAsset->GetVersion();
+                if (m_materialTypeVersion != materialTypeVersion)
```

Comment:
Required: we should do "<" here instead of "!=". There is no reason to support downgrading a material version. And in the future it won't even be possible to do so; I expect we'll eventually have upgrade steps that are not bidirectional, like "removeProperty" for example.